### DISCUSSION TITLE
Excessing HEAD requests for maven dependencies scanning

### DISCUSSION THREAD

**Discussion by devanjain** [renovatebot/renovate#11015] [Request Help]
**How are you running Renovate?**

- [ ] Self hosted

Please select which platform you are using:

- [ ] Bitbucket Server

Renovate version: "25.54.0"

**Describe the bug**

While scanning versions for MVN dependencies out of pom.xml in package manager, we are creating multiple requests to package manager.  In case of large number of repositories containing lot of dependencies causes load on package manager applications.

Lets say if the package manager has 100 release versions for the package, bot creates 102 requests for package.
1 GET request for metadata.xml
100 HEAD requests for each version
1 GET request for pom.xml from latest version

While for NPM packages, we create single request per package dependency, and for PYPI single request per pipfile. We should avoid making per version request in maven packages.

...

**Relevant debug logs**

**Comment by rarkins** [comment]
You can configure `RENOVATE_EXPERIMENTAL_NO_MAVEN_POM_CHECK=y` in env to skip these checks, but note that some local registries have been known to have corrupted indexes which can result in Renovate PRs proposing updates to phantom releases which have no download. This is why we added the check to make sure that ever release that's in the index actually has a corresponding download in a subdirectory.
